### PR TITLE
MAINT: Remove `f2py.f2py_testing` without replacement

### LINKED ISSUE #15528: Use of eval (eval-used)
**Use of eval (eval-used)**

Pylint throwing this warning:

[numpy/f2py/f2py_testing.py#L13](https://github.com/numpy/numpy/blob/master/numpy/f2py/f2py_testing.py#L13): W0123: Use of eval (eval-used)

### PR DESCRIPTION
**Pull request by HaoZeke**
Closes https://github.com/numpy/numpy/issues/15528.

The only `eval`s left lying around generate `lambda`s dynamically and will remain pending a rather major rewrite.

### CODE REVIEW

**Comment by seberg** [comment]
This looks good to me, even skipping release note.  (I am reading this right, that this looks like it was meant to help debugging while developing f2py, right?)
In the unlikely event that someone used it, vendoring should be easy enough.

@melissawm just put it in?

File: numpy/f2py/__init__.py [hunk 1]
```diff
@@ -190,11 +186,7 @@ def __getattr__(attr):
                                  "{!r}".format(__name__, attr))
 
     def __dir__():
-        return list(globals().keys() | {"f2py_testing", "test"})
+        return list(globals().keys())
```
Comment (seberg) [inline numpy/f2py/__init__.py:189]: Oh, "test" still exists.  Not sure it is worth hiding that here?

File: numpy/f2py/f2py_testing.py [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-                if diff_memusage2 != diff_memusage:
-                    print('memory usage change at step %i:' % i,
-                          diff_memusage2 - diff_memusage,
-                          fname)
-                    diff_memusage = diff_memusage2
-    current_memusage = memusage()
-    print('run', repeat * len(test_functions), 'tests',
-          'in %.2f seconds' % ((jiffies() - start_jiffies) / 100.0))
-    if start_memusage:
-        print('initial virtual memory size:', start_memusage, 'bytes')
```
Comment (HaoZeke) [inline numpy/f2py/f2py_testing.py:45]: @pearu noted that this might be useful for debugging and so could be added to the docs

Comment (seberg) [reply numpy/f2py/f2py_testing.py:45]: Yeah, on the other hand, there are more memory profilers these days...  And things like valgrind/pytest-leaks which are often more sensitive.  Having this type of information documented somewhere may be useful, but for myself, I would not worry about rescueing this code (I also don't mind rescueing it though).

**Comment by melissawm** [comment]
Any idea on the azure failures?

**Comment by HaoZeke** [comment]
> Any idea on the azure failures?

They should be gone now, they were part of the build failures introduced and fixed earlier, I'll force push to check again.

**Comment by seberg** [comment]
Sounds like we can put this in then now that CI passes.  Thanks Rohit and Melissa.

**Pull request merged**